### PR TITLE
added hugging face client

### PR DESCRIPTION
**Pull request by EyalPaz-700**

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -52,7 +52,11 @@ local = [
     "transformers >= 4.0.0"
 ]
 all = [
-    "llm-client[openai,ai21,local]"
+    "llm-client[openai,ai21,local,huggingface]"
+]
+huggingface = [
+    "llm-client[base-api]",
+    "transformers >= 4.0.0"
```
Comment (uripeled2) [inline pyproject.toml:59]: you need to add huggingface also to api

File: llm_client/llm_api_client/huggingface_client.py [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+TEXT_KEY = "generated_text"
+AUTH_HEADER = "Authorization"
+BEARER_TOKEN = "Bearer "
+DEFAULT_MODEL = "oasst-sft-4-pythia-12b-epoch-3.5"
+CONST_SLASH = '/'
+EMPTY_STR = ''
+NEWLINE = '\n'
+
+
+class HuggingFaceClient(BaseLLMAPIClient):
```
Comment (uripeled2) [inline llm_client/llm_api_client/huggingface_client.py:17]: can you add use of HuggingFaceClient in examples/only_llm_api_client.py

File: tests/llm_api_client/huggingface_client/test_huggingface.py [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+@pytest.mark.asyncio
+async def test_text_completion__sanity(mock_aioresponse, llm_client, url):
+    mock_aioresponse.post(
+        url,
+        payload=load_json_resource("huggingface/text_completion.json")
+    )
+
+    actual = await llm_client.text_completion(prompt="who is kobe bryant")
+
+    assert actual == ['who is kobe bryant?', 'Kobe Bryant is a retired professional basketball player who played for the Los Angeles Lakers of']
```
Comment (uripeled2) [inline tests/llm_api_client/huggingface_client/test_huggingface.py:23]: the prompt should not be in the response?

File: llm_client/llm_api_client/huggingface_client.py (hunk 2 above)
Comment (EyalPaz-700) [reply llm_client/llm_api_client/huggingface_client.py:17]: will do

File: pyproject.toml (hunk 1 above)
Comment (EyalPaz-700) [reply pyproject.toml:59]: will do

File: tests/llm_api_client/huggingface_client/test_huggingface.py (hunk 3 above)
Comment (EyalPaz-700) [reply tests/llm_api_client/huggingface_client/test_huggingface.py:23]: a "modified" prompt returns in the api output, i can remove it if needed but i am not sure that it's neccesary

File: pyproject.toml (hunk 1 above)
Comment (EyalPaz-700) [reply pyproject.toml:59]: done

File: tests/llm_api_client/huggingface_client/test_huggingface.py (hunk 3 above)
Comment (EyalPaz-700) [reply tests/llm_api_client/huggingface_client/test_huggingface.py:23]: done

**Pull request merged**